base/services/tftpd/CMakeLists.txt

```diff
@@ -1,9 +1,7 @@
 
-if(NOT MSVC)
-    add_compile_flags("-Wno-format-overflow")
-endif()
-
 add_executable(tftpd tftpd.cpp)
 set_module_type(tftpd win32cui)
 add_importlibs(tftpd advapi32 ws2_32 iphlpapi msvcrt kernel32)
 add_cd_file(TARGET tftpd DESTINATION reactos/system32 FOR all)
+
+target_compile_options(tftpd PRIVATE $<$<NOT:$<C_COMPILER_ID:MSVC>>:-Wno-format-overflow>)
```

Comment:
Does an old `if(NOT MSVC)` syntax have any limitations?
It feels more straightforward

Reply:
I guess it depends on who you are. I like the one-liner more.

Reply:
tbh this triple nested thing is not really readable.

Reply:
> Does an old `if(NOT MSVC)` syntax have any limitations?
> It feels more straightforward

It's also not the same: "MSVC" is true for both msvc and clang-cl, but C_COMPILER_ID:MSVC is true only for vanilla MSVC

Reply:
> tbh this triple nested thing is not really readable.

Then we can have it the other way around
$<$<C_COMPILER_ID:GNU,Clang>:-Wno-format-overflow> ?

Or we can have it like this:
```
set(C_COMPILER_NOT_MSVC $<NOT:$<C_COMPILER_ID:MSVC>>)
$<${C_COMPILER_NOT_MSVC}:-Wno-format-overflow>
```

Reply:
Of those, I find the `$<$<C_COMPILER_ID:GNU,Clang>:-Wno-format-overflow>` be the most readable.

Reply:
Please keep ```if (NOT MSVC)```. I don't see any reason to use this ugly and barely readable syntax unless explicitly needed for transitivity. Saving 2 lines of code doesn't fix anything.

Reply:
Done